Libraries/dotNetRDF/Query/Builder/Expressions/LiteralExpressionExtensions.cs

```diff
@@ -33,58 +33,37 @@ internal static class LiteralExpressionExtensions
     {
         private static readonly NodeFactory NodeFactory = new NodeFactory();
 
-        private static ILiteralNode ToLiteral<T>(T numericValue)
+        internal static ILiteralNode ToLiteral(object value)
         {
-            if (typeof(T) == typeof(int))
+            switch (value)
```

Comment:
Turned those ifs into C# 7.1 pattern matching of generic type argument. Do you have any objections? I understand that it only affect build-time requirements?

Reply:
No objection from me - I tend to update C# syntax as I go along in any case.